File: abstract-factory/go/src/vehicle/Motorcycle.go
package vehicle

import (
  "fmt"
)

// 具体产品实现类
type Motorcycle struct {
  Name string
}

func (m Motorcycle) GetName() string {
  return m.Name
}

func (m *Motorcycle) Run() {
  fmt.Println("Motorcycle::Run() [Name=" + m.Name + "]")
}

File: abstract-factory/c/src/vehicle/motorcycle.c
#include "../func.h"

/* Motorcycle对象函数实现 */
void motorcycle_run(Vehicle *vehicle)
{
  printf("\r\n motorcycle_run() [name=%s]", vehicle->name);
}

/* 创建对象构造函数 */
Motorcycle *motorcycle_constructor(char *name)
{
  Motorcycle *motorcycle = (Motorcycle *)malloc(sizeof(Motorcycle));
  strncpy(motorcycle->name, name, 50);
  motorcycle->run = &motorcycle_run;
  return motorcycle;
}